### ISSUE TITLE
Bug: Can't reconnect IMAP account

### ISSUE THREAD

**Issue by hummerd** [Foundry376/Mailspring#1814]
**Describe the bug**
When I click reconnect empty dialog box appears.

**To Reproduce**
My password for email account (IMAP for MS Exchange) has expired. When I try to reconnect account - empty dialog box appears.

**Expected behavior**
Non empty dialog box.

**Screenshots**
![mail-err](https://user-images.githubusercontent.com/1921665/75272918-ddf81c00-580f-11ea-9277-b5c46f8068fe.png)

**Setup: (please complete the following information):**
 - OS: Ubuntu
 - OS Version: 19.10
 - Mailspring Version: 1.7.4-d54

**Additional context** 
Some logs
```
{ Error: Attempting to toJSON AttributeDateTime which is not a date: authedAt = 1970-01-01T00:00:00.000Z
    at AttributeDateTime.toJSON (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/attributes/attribute-datetime.ts:15:13)
    at Account.toJSON (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/models/model.ts:115:40)
    at Account.toJSON (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/models/account.ts:126:24)
    at Account.clone (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/models/model.ts:82:38)
    at new Composed (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/internal_packages/onboarding/lib/decorators/create-page-for-form.tsx:36:39)
    at Ag (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:135:201)
    at $g (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:157:149)
    at eh (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:171:13)
    at ai (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:202:255)
    at bi (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:203:271)
    at Fi (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:218:174)
    at Gi (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:217:174)
    at ii (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:214:252)
    at xg (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:212:94)
    at Object.enqueueSetState (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:133:223)
    at InjectedComponentSet.Component.setState (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react/cjs/react.development.js:373:16)
    at ComponentRegistry._componentUnlistener.mailspring_exports_1.ComponentRegistry.listen (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/components/injected-component-set.tsx:114:12)
    at EventEmitter.eventHandler (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/global/mailspring-store.ts:128:23)
    at EventEmitter.emit (events.js:187:15)
    at EventEmitter.emit (domain.js:442:20)
    at ComponentRegistry.trigger (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/global/mailspring-store.ts:141:26)
    at ComponentRegistry.triggerDebounced.underscore_1.default.debounce (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/registries/component-registry.ts:197:44)
    at later (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/underscore/underscore.js:828:25)
  message:
   'Attempting to toJSON AttributeDateTime which is not a date: authedAt = 1970-01-01T00:00:00.000Z' } { pluginIds: [ 'onboarding' ] }
{ Error: Attempting to toJSON AttributeDateTime which is not a date: authedAt = 1970-01-01T00:00:00.000Z
    at AttributeDateTime.toJSON (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/attributes/attribute-datetime.ts:15:13)
    at Account.toJSON (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/models/model.ts:115:40)
    at Account.toJSON (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/models/account.ts:126:24)
    at Account.clone (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/models/model.ts:82:38)
    at new Composed (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/internal_packages/onboarding/lib/decorators/create-page-for-form.tsx:36:39)
    at Ag (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:135:201)
    at $g (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:157:149)
    at eh (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:171:13)
    at ai (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:202:255)
    at bi (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:203:271)
    at Fi (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:218:174)
    at Gi (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:217:174)
    at ii (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:214:252)
    at xg (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:212:94)
    at Object.enqueueSetState (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js:133:223)
    at InjectedComponentSet.Component.setState (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react/cjs/react.development.js:373:16)
    at ComponentRegistry._componentUnlistener.mailspring_exports_1.ComponentRegistry.listen (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/components/injected-component-set.tsx:114:12)
    at EventEmitter.eventHandler (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/global/mailspring-store.ts:128:23)
    at EventEmitter.emit (events.js:187:15)
    at EventEmitter.emit (domain.js:442:20)
    at ComponentRegistry.trigger (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/global/mailspring-store.ts:141:26)
    at ComponentRegistry.triggerDebounced.underscore_1.default.debounce (file:///tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/registries/component-registry.ts:197:44)
    at later (/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/underscore/underscore.js:828:25)
  message:
   'Attempting to toJSON AttributeDateTime which is not a date: authedAt = 1970-01-01T00:00:00.000Z' } { url:
   '/snap/mailspring/440/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.production.min.js',
  line: 217,
  column: 372,
  pluginIds: [ 'onboarding' ] }
Raven: 429 - undefined
Gtk-Message: GtkDialog mapped without a transient parent. This is discouraged.

```

**Comment by AlexDunmow** [comment]
Potential dupe of my bug? #1811

**Comment by BilouMaster** [comment]
Hi, I tested Mailspring for the first time yesterday and I have the same problem but when running the app for the first time and setting up a personal imap account for the first email connexion. (The application never starts, I just keep getting the empty dialog box)

The empty dialog box comes quick after a "success" dialog box.

Yesterday I tested on another computer and started using Mailspring with two gmail accounts, it worked like a charm but when I added the imap account, it showed the "success" dialog box but my imap account wasn't added to the list (neither in the mailbox nor the accounts gestion list in settings)

I need to get that imap account working (mine is provided by one.com, which is listed on getmailspring.com) so "Mailspring" is not for me until the next bugfix if any.

**Mailspring version:** 1.7.4
**OS:** Windows 7 and Windows 10

**Comment by hummerd** [comment]
> Potential dupe of my bug? #1811

I think it is different one. #1811 says about problem after entering credentials, while this one is about that I am not able to enter them at all.

**Comment by timschmele** [comment]
Exacty the same issue here, same Mailspring version, different OS!

OS: OSX
OS Version: 10.13.6
Mailspring Version: 1.7.4-d54

**Comment by bengotow** [comment]
Hey folks! Thanks for reporting this and submitting a fix - I'm sorry it's taken me so long to ship a build with the changes. I honestly can't believe it's April already. I'm doing QA tonight and I'll ship the 1.7.5 release with the patch ASAP tomorrow!

**Comment by fracsi** [comment]
When trying to update connection setting for an imap account the error is still present in:
```
OS: Ubuntu
OS Version: 19.10
Mailspring version: 1.7.5-a97
```

Error in console:
```
Error: Attempting to toJSON AttributeDateTime which is not a date: authedAt = 2020-04-17T06:26:10.936Z
    at AttributeDateTime.toJSON (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/attributes/attribute-datetime.ts:15)
    at Account.toJSON (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/models/model.ts:115)
    at Account.toJSON (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/models/account.ts:126)
    at Account.clone (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/flux/models/model.ts:82)
    at new Composed (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/internal_packages/onboarding/lib/decorators/create-page-for-form.tsx:36)
    at constructClassInstance (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:11864)
    at updateClassComponent (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:13681)
    at beginWork (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:14489)
    at performUnitOfWork (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:17014)
    at workLoop (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:17054)
    at HTMLUnknownElement.callCallback (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:149)
    at Object.invokeGuardedCallbackDev (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:199)
    at invokeGuardedCallback (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:256)
    at replayUnitOfWork (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:16366)
    at renderRoot (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:17148)
    at performWorkOnRoot (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:17993)
    at performWork (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:17901)
    at performSyncWork (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:17873)
    at requestWork (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:17761)
    at scheduleWork (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:17566)
    at Object.enqueueSetState (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react-dom/cjs/react-dom.development.js:11695)
    at InjectedComponentSet.Component.setState (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/react/cjs/react.development.js:373)
    at ComponentRegistry._componentUnlistener.mailspring_exports_1.ComponentRegistry.listen (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/components/injected-component-set.tsx:114)
    at EventEmitter.eventHandler (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/global/mailspring-store.ts:128)
    at EventEmitter.emit (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/static/events.js:187)
    at EventEmitter.emit (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/static/domain.js:442)
    at ComponentRegistry.trigger (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/global/mailspring-store.ts:141)
    at ComponentRegistry.triggerDebounced.underscore_1.default.debounce (/tmp/nylas-build/electron-packager/linux-x64/mailspring-linux-x64/resources/app/src/registries/component-registry.ts:197)
    at later (/snap/mailspring/456/usr/share/mailspring/resources/app.asar/node_modules/underscore/underscore.js:828) Object
```

**Comment by westr029** [comment]
Having the same problem here with reconnecting to MS outlook 365:

OS: Ubuntu 20.04
mailspring version: 1.7.5-a976d710

Might be related to #1910, #1817 and closed issue #1864 and #1861

**Comment by bengotow** [comment]
Hey folks, thanks for reporting this. Happy to say this has been resolved in 1.7.6 which shipped this afternoon. 🚀  The app should auto-update, but if it doesn't you can also grab the new release from GitHub's Releases tab or from https://getmailspring.com/download. Thanks for your patience and help tracking this down, enjoy the update!
